File: 128-longest-consecutive-sequence/longest-consecutive-sequence-better.cpp
// Better - Approach => (Using Sorting)


class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int cnt = 1;
        int maxcnt = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] - nums[i - 1] == 1) {
                cnt++;
                maxcnt = max(maxcnt, cnt);
            } 
            else if (nums[i] == nums[i - 1]){
                continue;
            }
            else {
                cnt = 1;
            }
        }
        return maxcnt;
    }
};



// T.C => O(n*log(n)) + O(n) = O(n*log(n))
// S.C => O(1)

File: 128-longest-consecutive-sequence/longest-consecutive-sequence-better.java
// Better Solution

class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        Arrays.sort(nums);
        int cnt = 1;
        int maxcnt = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - nums[i - 1] == 1) {
                cnt++;
                maxcnt = Math.max(maxcnt, cnt);
            }
            else if (nums[i] - nums[i - 1] == 0){
                continue;
            }
            else {
                cnt = 1;
            }
        }
        return maxcnt;
    }
}

// T.C => O(nlog(n))
// S.C => O(1)
